src/test/java/org/jboss/aerogear/keycloak/metrics/PrometheusExporterTest.java

```diff
@@ -0,0 +1,54 @@
+package org.jboss.aerogear.keycloak.metrics;
+
+import org.hamcrest.MatcherAssert;
+import org.junit.Test;
+import org.keycloak.events.Event;
+import org.keycloak.events.EventType;
+import org.keycloak.events.admin.OperationType;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+public class PrometheusExporterTest {
+  private Event createEvent(EventType type) {
+    Event event = new Event();
+    event.setType(type);
+    event.setRealmId("myrealm");
+    return event;
+  }
+
+  @Test
+  public void shouldRegisterAllKeycloakEvents() {
+    int userEvents = EventType.values().length;
+    int adminEvents = OperationType.values().length;
+
+    MatcherAssert.assertThat(
+      "All events registered",
+      userEvents + adminEvents == PrometheusExporter.counters.size());
+  }
+
+  @Test
+  public void shouldCorrectlyCountLoginsAndLogouts() throws IOException {
+    Event login = createEvent(EventType.LOGIN);
+    Event impersonate = createEvent(EventType.IMPERSONATE);
```

Comment:
I think here 1 extra thing is being tested. Since the impersonation being considered as a login is a custom implementation. Maybe we should use here only Login event and create a new test `shouldCountImpersonateEventLikeALogin`.